internal/clients/http/ec2/iam_client.go

```diff
@@ -0,0 +1,304 @@
+package ec2
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	iamTypes "github.com/aws/aws-sdk-go-v2/service/iam/types"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/RHEnVision/provisioning-backend/internal/clients/http"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+)
+
+type Statement struct {
+	Sid    string
+	Effect string
+	Action []string
+}
+
+type Policy struct {
+	Statement []Statement `json:"Statement"`
+}
+
```

Comment:
I think documentation is missing here as we do for other types: https://github.com/RHEnVision/provisioning-backend/blob/20294330f9b82e703c799d7e87643e263210c816/internal/clients/instance_type.go#L17

Reply:
I think this is a valid point. Even though the structure can be seen below, it might be a good practice. I've left a few comments.